modules/Bio/EnsEMBL/Hive/HivePipeline.pm

```diff
@@ -544,151 +544,203 @@ sub print_diagram {
 sub apply_tweaks {
     my $self    = shift @_;
     my $tweaks  = shift @_;
-
+    my @responce;
+    my $responceStructure;
     my $need_write = 0;
-
+    $responceStructure->{Tweaks} = [];
     foreach my $tweak (@$tweaks) {
-        print "\nTweak.Request\t$tweak\n";
+        push @responce, "Tweak.Request  \t$tweak\n";
 
         if($tweak=~/^pipeline\.param\[(\w+)\](\?|#|=(.+))$/) {
             my ($param_name, $operator, $new_value_str) = ($1, $2, $3);
-
+            my $tweakStructure;
```

Comment:
Since this will be used in all cases of the `if`, can be definition be moved a level up ?
Also, to avoid auto-vivification, can we at least set it to an empty hash ?

Reply:
We need every iteration be sure that $tweakStructure in new (doesn't contain any previous result), so I create it every time locally;